wallet/src/multisig_account.rs

```diff
@@ -0,0 +1,232 @@
+use curve25519_dalek::EdwardsPoint;
+use nimiq_hash::Blake2bHasher;
+use nimiq_keys::{
+    multisig::{hash_public_keys, Commitment, CommitmentPair, PartialSignature, RandomSecret},
+    Address, KeyPair, PublicKey, SecureGenerate, SecureRng, Signature,
+};
+use nimiq_primitives::{coin::Coin, networks::NetworkId};
+use nimiq_serde::Serialize;
+use nimiq_transaction::{SignatureProof, Transaction};
+use nimiq_utils::merkle::{compute_root_from_content_slice, MerklePath};
+
+pub struct MultiSigAccount {
+    pub address: Address,
+    pub key_pair: KeyPair,
+    pub min_signatures: u8,
+    pub public_keys: Vec<PublicKey>,
+}
+
+impl MultiSigAccount {
+    /// Returns a new MultiSignature Account.
+    ///
+    /// # Arguments
+    ///
+    /// * `key_pair` - KeyPair owning this account.
+    /// * `min_signatures` - Number of signatures required. Panics if value is less than 1.
+    /// * `public_keys` - A list of all owners' public keys. Panics if list is missing the public key of `key_pair`.
+    pub fn from_public_keys(
+        key_pair: &KeyPair,
+        min_signatures: u8,
+        public_keys: &[PublicKey],
+    ) -> Self {
+        assert!(
+            !public_keys.is_empty(),
+            "Provided public keys may not be empty."
+        );
+        assert!(
+            min_signatures > 0,
+            "Minimal signatures must be greater than 0."
+        );
+        assert!(
+            public_keys.contains(&key_pair.public),
+            "Public key of keypair must be part of provided public keys."
+        );
+
+        let mut sorted_public_keys = Vec::with_capacity(public_keys.len());
+        public_keys.clone_into(&mut sorted_public_keys);
+        sorted_public_keys.sort();
```

Comment:
I think we should be able to do:
```suggestion
        let mut sorted_public_keys = public_keys.clone();
        sorted_public_keys.sort();
```

Reply:
By calling `public_keys.clone()` you clone the reference instead of the actual values. So each item has to be cloned.
I've come up with:

```
let mut sorted_public_keys: Vec<PublicKey> = public_keys.iter().cloned().collect();
sorted_public_keys.sort();
```

Reply:
Mmm... I think the problem is that you want a mutable copy of the array elements which can't be done with `clone`. But then I suggest to do:
```
let mut sorted_public_keys = public_keys.to_vec();
sorted_public_keys.sort()
```